searx/engines/pdbe.py

```diff
@@ -0,0 +1,118 @@
+# link to preview image of structure
+pdbe_preview_url = 'https://www.ebi.ac.uk/pdbe/static/entry/{pdb_id}_deposited_chain_front_image-200x200.png'
+
+
+def request(query, params):
+
+    params['url'] = pdbe_solr_url
+    params['method'] = 'POST'
+    # using searx User-Agent
+    params['headers']['User-Agent'] = searx_useragent()
+    params['data'] = {
+        'q': "{}".format(query),
+        'wt': "json"  # request response in parsable format
+    }
+    return params
+
+
+def construct_body(result):
+    # set title
+    title = result['title']
+
+    # construct content body
+    content = """{title}<br />{authors} {journal} <strong>{volume}</strong>&nbsp;{page} ({year})"""
+
+    # replace placeholders with actual content
+    try:
+        if result['journal']:
+            content = content.format(
+                title=result['citation_title'],
+                authors=result['entry_author_list'][0], journal=result['journal'], volume=result['journal_volume'],
+                page=result['journal_page'], year=result['citation_year'])
+        else:
+            content = content.format(
+                title=result['citation_title'],
+                authors=result['entry_author_list'][0], journal='', volume='', page='', year=result['release_year'])
+        img_src = pdbe_preview_url.format(pdb_id=result['pdb_id'])
+    except (KeyError):
+        content = None
+        img_src = None
+
+    # construct url for preview image
+    try:
+        img_src = pdbe_preview_url.format(pdb_id=result['pdb_id'])
+    except (KeyError):
+        img_src = None
+
+    return [title, content, img_src]
+
+
+def response(resp):
+
+    results = []
+    json = loads(resp.text)['response']['docs']
+
+    # parse results
+    for result in json:
+        # reset skip flag
+        skip_result = False
+
+        # catch obsolete entries and mark them accordingly
+        if result['status'] in pdb_unpublished_codes:
+            skip_result = True
+        elif result['status'] == 'OBS':
+            if not hide_obsolete:
+                # expand title to add some sort of warning message
+                title = gettext('{title}&nbsp;(OBSOLETE)').format(title=result['title'])
+                superseded_url = pdbe_entry_url.format(pdb_id=result['superseded_by'])
+
+                # since we can't construct a proper body from the response, we'll make up our own
+                msg_superseded = gettext("This entry has been superseded by")
+                content = '<em>{msg_superseded} \<a href="{url}">{pdb_id}</a></em>'.format(
+                    msg_superseded=msg_superseded,
+                    url=superseded_url,
+                    pdb_id=result['superseded_by'], )
+
+                # obsoleted entries don't have preview images
+                img_src = None
+            else:
+                skip_result = True
+        else:
+            title, content, img_src = construct_body(result)
+
+        if not skip_result:
```

Comment:
This `if` is not required either, because when `continue` is used, processing never gets here if the result has to be skipped.